### PR TITLE
feat: APPS-2865 Create EventSeries detail page / slug template to verify gql query data

### PR DESCRIPTION
**Pull request by jendiamond**
Connected to [APPS-2865](https://jira.library.ucla.edu/browse/APPS-2865)

https://deploy-preview-31--test-ftva.netlify.app/series/step-up-series

**Page/Pages Created*
+ `pages/series/[...slug].vue`
+ `gql/queries/FTVAEventSeriesDetail.gql`

**Notes:**
Changed the query name from   
`gql/queries/FTVAEventDetailSeries.gql`
to. 
`gql/queries/FTVAEventSeriesDetail.gql`

Screenshot 2024-08-28 at 3 17 55 PM

[APPS-2865]: https://uclalibrary.atlassian.net/browse/APPS-2865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pages/events/[...slug].vue
Comment (pghorpade) [inline pages/events/[...slug].vue:1]: @jendiamond I dont think we should be making this a catch all slug page, @axamei is event page going to be like general content pages now?

File: pages/series/[...slug].vue
Comment (pghorpade) [inline pages/series/[...slug].vue:1]: please revert the file name if eventseries are not catch all pages like general content pages.

File: gql/queries/FTVAEventSeriesDetail.gql [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    eventTitle
+    startDateWithTime
+    ftvaImage {
+      ...Image
+    }
+    ftvaEventFilters {
+      title
+    }
+  }
+  otherSeriesOngoing: entries(section: "ftvaEventSeries", limit: 3, ongoing: true) {
```
Comment (pghorpade) [inline gql/queries/FTVAEventSeriesDetail.gql:71]: @jendiamond  Add orderby to this upcommingevents query part like `otherSeriesOngoing: entries(section: "ftvaEventSeries", limit: 3, ongoing: true, orderBy: "startDateWithTime ASC")`

**Review by pghorpade — CHANGES_REQUESTED** [review]

File: pages/series/[...slug].vue [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+</script>
+
+<template>
+  <main
+    id="main"
+    class="page page-event-series-detail"
+  >
+    <div class="one-column">
+      <NavBreadcrumb
+        class="breadcrumb"
```
Comment (pghorpade) [inline pages/series/[...slug].vue:98]: pass another prop parent-title="Screening Series"

File: pages/events/[...slug].vue
Comment (jendiamond) [reply pages/events/[...slug].vue:1]: change it to [slug].vue right

**Review by tinuola — APPROVED** [review]

File: pages/[...slug].vue [hunk 3]
```diff
@@ -132,7 +132,10 @@ onMounted(() => {
   </main>
```
[the remark this answers is not in the document]
Comment (pghorpade) [reply pages/[...slug].vue:132]: This is the rename you made in this PR. [pages/[...slug].vue.bak → pages/[...slug].vue], revert this.

File: pages/series/[slug].vue [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+
+const page = ref(_get(data.value, 'ftvaEventSeries', {}))
+const upcomingEvents = ref(_get(data.value, 'upcomingEvents', {}))
+const pastEvents = ref(_get(data.value, 'pastEvents', {}))
+const otherSeriesOngoing = ref(_get(data.value, 'otherSeriesOngoing', {}))
+const otherSeriesUpcoming = ref(_get(data.value, 'otherSeriesUpcoming', {}))
+
+watch(data, (newVal, oldVal) => {
+  console.log('In watch preview enabled, newVal, oldVal', newVal, oldVal)
+  page.value = _get(data.value, 'ftvaEventSeries', {})
```
Comment (pghorpade) [inline pages/series/[slug].vue:44]: replace data.value with newVal on this line, newVal is not a ref

File: pages/series/[slug].vue [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+  upcomingEvents.value = ref(_get(data.value, 'upcomingEvents', {}))
```
Comment (pghorpade) [inline pages/series/[slug].vue:45]: replace data.value with newVal on this line, newVal is not a ref

File: pages/series/[slug].vue [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+  pastEvents.value = ref(_get(data.value, 'pastEvents', {}))
```
Comment (pghorpade) [inline pages/series/[slug].vue:46]: replace data.value with newVal on this line, newVal is not a ref

File: pages/series/[slug].vue [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+  otherSeriesOngoing.value = ref(_get(data.value, 'otherSeriesOngoing', {}))
```
Comment (pghorpade) [inline pages/series/[slug].vue:47]: replace data.value with newVal on this line, newVal is not a ref

File: pages/series/[slug].vue [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+  otherSeriesUpcoming.value = ref(_get(data.value, 'otherSeriesUpcoming', {}))
```
Comment (pghorpade) [inline pages/series/[slug].vue:48]: replace data.value with newVal on this line, newVal is not a ref

File: pages/series/[slug].vue (hunk 4 above)
Comment (pghorpade) [reply pages/series/[slug].vue:44]: make similar change in event slug page on preview code

File: pages/series/[slug].vue [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+})
+
+// Transform data for Carousel
+const parsedCarouselData = computed(() => {
+  // map image to item, map creditText to credit
+  return parsedImage.value.map((rawItem, index) => {
+    return {
+      item: [{ ...rawItem.image[0], kind: 'image' }], // Carousels on this page are always images, no videos
+      credit: rawItem?.creditText,
+      captionTitle: 'dfdsfs', // TODO do we need these? test without
```
Comment (pghorpade) [inline pages/series/[slug].vue:63]: check with @axamei  if we need captiontitle and captionText

File: pages/series/[slug].vue [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+      captionText: 'dfsdfsd',
+    }
+  })
+})
+
+const parsedFtvaEventSeries = computed(() => {
+  /* Early Returns: If series.value is falsy or empty, or
+    if firstSeries.ftvaEvent is falsy or empty, return
+    an empty array immediately. */
+  if (!series.value || series.value.length === 0) {
```
Comment (pghorpade) [inline pages/series/[slug].vue:73]: Remove this computed method as there is no series variable in this file. and logic for explore more series is going to be different

File: pages/series/[slug].vue (hunk 9 above)
Comment (jendiamond) [reply pages/series/[slug].vue]: we don't need them

File: pages/series/[slug].vue (hunk 4 above)
Comment (pghorpade) [reply pages/series/[slug].vue:44]: this is still not updated page.value = _get(data.value, 'ftvaEventSeries', {})

**Review by pghorpade — CHANGES_REQUESTED** [review]

**Review by pghorpade — APPROVED** [review]
👍 Good work

**Pull request merged**